Apps/ClinicalDocument/test/unit/Services/ClinicalDocumentServiceTests.cs

```diff
@@ -102,22 +116,60 @@ public async Task ShouldGetClinicalDocumentRecordsThrowsException()
         /// <summary>
         ///  Get clinical document file - happy path.
         /// </summary>
+        /// <param name="canAccessDatasource">The value indicates whether the clinical document data source can be accessed or not.</param>
+        /// <param name="personalAccountExists">
+        /// The bool value indicating whether a pid exists or not to get the clinical document
+        /// records.
+        /// </param>
+        /// <param name="personalAccountResultType">The result type for personal account response.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
-        [Fact]
-        public async Task ShouldGetClinicalDocumentFile()
+        [Theory]
+        [InlineData(true, true, ResultType.Success)]
+        [InlineData(true, true, ResultType.Error)]
+        [InlineData(true, false, ResultType.Success)]
+        [InlineData(false, true, null)]
+        public async Task ShouldGetClinicalDocumentFile(bool canAccessDatasource, bool personalAccountExists, ResultType? personalAccountResultType)
         {
             // Arrange
             Guid id = Guid.NewGuid();
             PhsaHealthDataResponse expectedPhsaHealthDataResponse = GetPhsaHealthDataResponse(id);
 
-            IClinicalDocumentService clinicalDocumentService = GetClinicalDocumentService(expectedPhsaHealthDataResponse, false);
+            IClinicalDocumentService clinicalDocumentService = GetClinicalDocumentService(expectedPhsaHealthDataResponse, false, canAccessDatasource, personalAccountExists, personalAccountResultType);
 
-            // Act
-            RequestResult<EncodedMedia> actualResult = await clinicalDocumentService.GetFileAsync(It.IsAny<string>(), It.IsAny<string>());
+            if (canAccessDatasource)
+            {
+                if (personalAccountExists)
+                {
+                    // Act
+                    RequestResult<EncodedMedia> actualResult = await clinicalDocumentService.GetFileAsync(It.IsAny<string>(), It.IsAny<string>());
```

Comment:
I think `It.IsAny` only makes sense in the context of a `.Setup()` call.

```suggestion
                    RequestResult<EncodedMedia> actualResult = await clinicalDocumentService.GetFileAsync(string.Empty, string.Empty);
```